fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -180,9 +180,123 @@ def additional_upload_parameters
     end
   end
 
+  # Generates commands and executes the altool.
+  class AltoolTransporterExecutor < TransporterExecutor
+    ERROR_REGEX = /\sNSLocalizedFailureReason\s=\s"+(.+)"/
+
+    private_constant :ERROR_REGEX
+
+    def execute(command, hide_output)
+      if Helper.test?
+        yield(nil) if block_given?
+        command
+      end
+
+      @errors = []
+      @warnings = []
+      @all_lines = []
+
+      if hide_output
+        # Show a one time message instead
+        UI.success("Waiting for App Store Connect transporter to be finished.")
+        UI.success("Application Loader progress... this might take a few minutes...")
+      end
+
+      begin
+        exit_status = FastlaneCore::FastlanePty.spawn(command) do |command_stdout, command_stdin, pid|
+          begin
+            command_stdout.each do |line|
+              @all_lines << line
+              parse_line(line, hide_output) # this is where the parsing happens
+            end
+          end
+        end
+      rescue => ex
+        # FastlanePty adds exit_status on to StandardError so every error will have a status code
+        exit_status = ex.exit_status
+        @errors << ex.to_s
+      end
+
+      @errors << "The call to the altool completed with a non-zero exit status: #{exit_status}. This indicates a failure." unless exit_status.zero?
+
+      UI.important(@warnings.join("\n")) unless @warnings.empty?
+
+      if @errors.count > 0 && @all_lines.count > 0
+        # Print out the last 18 lines, this is key for non-verbose mode
+        @all_lines.last(18).each do |line|
+          UI.important("[altool] #{line}")
+        end
+        UI.message("Application Loader output above ^")
+        UI.error(@errors.join("\n"))
+      end
+
+      yield(@all_lines) if block_given?
+      exit_status.zero?
+    end
+
+    def build_upload_command(username, password, source = "/tmp", provider_short_name = "", jwt = nil, platform = nil, api_key = nil)
+      [
+        ("API_PRIVATE_KEYS_DIR=#{api_key[:key_filepath]}" if api_key),
```

Comment:
- we are expecting `key_filepath` which will break all the CI because app_store_connect_api_key action supports api key_content inside the ENV variable `(the content of the .p8 file)` also instead of passing the .p8 file-path. 
- Please see the last part of this document https://docs.fastlane.tools/app-store-connect-api/
- i think, we should file a way to handle that scenario also. maybe we need to create .p8 file 🙈
- for example: **app_store_connect_api_key** accepts the `key_content` instead of `p8 file-path`
```ruby
default_platform(:ios)

platform :ios do
  desc "Description of what the lane does"
  lane :custom_lane do
    xcode_select("/Applications/Xcode_14 beta 6.app")
    api_key = app_store_connect_api_key(
      key_id: "XXXXXX",
      issuer_id: "XXX-XXX-XXXX",
      key_content: "-----BEGIN PRIVATE KEY-----\nXXXXXXX\nXXXXXX\nXXXXXXXX\nXXX\n-----END PRIVATE KEY-----",
      duration: 1200,
      in_house: false
    )    
    
    pilot(api_key: api_key)
  end
end

Reply:
Ah... I missed this case, thank you!

Reply:
> i think, we should find a way to handle that scenario also. maybe we need to create .p8 file from key_content 🙈

yes! it is solution for this case, let me do so

Reply:
I noticed key_filepath is not existed and i need to do other way, I'll push commit soon

Reply:
https://github.com/fastlane/fastlane/pull/20631/commits/747178f1ed4cd5866a218f2eecfedfb1dd68d119 fixed